rinja_derive/src/input.rs

```diff
@@ -149,7 +152,21 @@ impl TemplateInput<'_> {
         let mut dependency_graph = Vec::new();
         let mut check = vec![(Arc::clone(&self.path), source, source_path)];
         while let Some((path, source, source_path)) = check.pop() {
-            let parsed = self.syntax.parse(source, source_path)?;
+            let parsed = match self.syntax.parse(Arc::clone(&source), source_path) {
+                Ok(parsed) => parsed,
+                Err(err) => {
+                    let msg = err
+                        .message
+                        .unwrap_or_else(|| "failed to parse template source".into());
+                    let file_path = err
+                        .file_path
+                        .as_deref()
+                        .unwrap_or(Path::new("<source attribute>"));
```

Comment:
If we have the `source` attribute span, can't we use it instead?

Reply:
To underline `source` or the value of `source` instead of the `template` token? I think this could get confusion if the error happens in a transcluded file.

Reply:
Hum... I think we can know if it comes from a file included. So there are two cases:

1. From the source directly, then we can underline the `source` attribute saying "it comes from template from this attribute"
2. From a file included in the source, then we still underline the `source` attribute and we say "it comes from a template included from this attribute"

What do you think?

Reply:
Yes, makes sense. Changed.